src/ScottPlot5/ScottPlot5 Tests/Unit Tests/UnitTests/IDataSourceTests.cs

```diff
@@ -0,0 +1,393 @@
+﻿using FluentAssertions;
+using ScottPlot.DataSources;
+
+namespace ScottPlotTests.UnitTests;
+
+internal class IDataSourceTests
+{
+    // Data that tests should use when creating the IDataSource
+
+    #region < Setup >
+
+    public static readonly double[] Xs = Generate.Consecutive(51, Period, 0);
```

Comment:
These tests are far too complicated for the Unit name, but I guess they do the trick.

Reply:
If you have any suggestions, we can do them. I wanted to standard the tests as much as possible, which is why I wrote them this way. All have same values used, call same assertions; all the signal sources get tested the same way, scatter, etc. while complex, this tests each IDataSource and IGetNearest on each implementation.

Reply:
We could also move them around. Put the scatter specific tests over in the scatter file, move the signal and signalXY into their own files, But then to call the method in the file would mean that work on the unit test you're looking at two separate files, which I wanted to avoid. There is an argument for both here. 

since I wanted the actual test of the interface to happen the same way every time, This is what I wound up with.

Reply:
I'm not arguing, you have a whole subsystem of tests that work. It's a matter of naming their unit. Never mind.